backend/main.py

```diff
@@ -300,7 +289,6 @@ async def get_function_call_response(
     template,
     task_model_id,
     user,
-    model,
```

Comment:
`model` overwritten on line 340 (line 328 in the updated version) and never used prior to that point